src/approaches/interactive_learning_approach.py

```diff
@@ -21,86 +24,106 @@ class InteractiveLearningApproach(NSRTLearningApproach):
-            else:  # No policy found
-                raise ApproachFailure("Failed to sample a task that approach "
-                                      "can solve.")  # pragma: no cover
-            # Roll out policy via a temporary hack
-            assert isinstance(self._option_model, _OracleOptionModel)
-            simulator = self._option_model._simulator  # pylint:disable=protected-access
-            traj, _, _ = utils.run_policy_on_task(
-                policy,
-                task,
-                simulator,
-                max_num_steps=CFG.interactive_max_num_steps)
-            # Decide whether to ask about each possible atom during exploration
-            for s in traj.states:
-                ground_atoms = utils.all_possible_ground_atoms(
-                    s, self._predicates_to_learn)
-                for atom in ground_atoms:
-                    # Note: future score functions will use the state s
-                    score = score_atom(dataset, atom)
-                    # Ask about this atom if it is the best seen so far
-                    if score > best_score:
-                        if self._ask_teacher(s, atom):
-                            # Add this atom if it's a positive example
-                            traj = LowLevelTrajectory([s], [])
-                            dataset.append(traj, [{atom}])
-                            # Still need a way to use negative examples
-                        best_score = score
-            if i % CFG.interactive_relearn_every == 0:
-                self._relearn_predicates_and_nsrts(dataset,
-                                                   online_learning_cycle=i - 1)
+        # Save dataset, to be used for online interaction
+        self._dataset = dataset
+
+    def get_interaction_requests(self) -> List[InteractionRequest]:
+        # Sample a train task
+        train_task_idx = self._rng.choice(len(self._train_tasks))
+        init = self._train_tasks[train_task_idx].init
+        # Detect and filter out static predicates
+        static_preds = utils.get_static_preds(
+            self._nsrts, self._get_current_predicates())
+        preds = self._get_current_predicates() - static_preds
+        # Find acting policy for the request
+        task_list = glib_sample(init, preds, self._dataset)
+        assert task_list
+        task, act_policy = self._find_first_solvable(task_list)
+        assert task.init is init
+        def _query_policy(s: State) -> Optional[GroundAtomHoldsQuery]:
+            # Decide whether to ask about each possible atom
+            ground_atoms = utils.all_possible_ground_atoms(
+                s, self._predicates_to_learn)
+            for atom in ground_atoms:
+                score = score_atom(self._dataset, atom)
+                # Ask about this atom if it is the best seen so far
+                if score >= self._best_score:
+                    # TODO: the previous line is a > on master, changed to >=
```

Comment:
this is a significant functional change, but since this score function is just a placeholder anyway, I think this change is okay. it will certainly lead to more interesting and testable behavior, where queries continue after the first state. @amburger66 let us know what you think though.

Reply:
sounds good!

Reply:
cool. we'll still do this in a separate PR if we do it, to keep this one a pure refactor.